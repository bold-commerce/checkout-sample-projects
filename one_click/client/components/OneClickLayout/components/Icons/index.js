export { default as TimesCircle } from './TimesCirlce'
export { default as ArrowRight } from './ArrowRight'
export { default as CaretRight } from './CaretRight'