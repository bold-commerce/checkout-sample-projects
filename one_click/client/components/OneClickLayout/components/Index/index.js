export { default as IndexPage } from "./IndexPage";
export { default as IndexGuest } from "./IndexGuest";