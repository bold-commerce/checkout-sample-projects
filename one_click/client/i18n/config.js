import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;