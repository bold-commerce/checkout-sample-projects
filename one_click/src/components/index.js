export * from './Address';
export * from './BackButton';
export * from './Billing';
export * from './BillingAddress';
export * from './Card';
export * from './CheckoutButton';
export * from './CheckoutForm';
export * from './Confirmation';
export * from './CustomerInfo';
export * from './Discount';
export * from './EmptyState';
export * from './Header';
export * from './Icons';
export * from './Index';
export * from './InventoryItem';
export * from './LineItems';
export * from './LoadingState';
export * from './Payment';
export * from '../pages/Processing';
export * from './Product';
export * from './RadioItem';
export * from './Shipping';
export * from './ShippingLines';
export * from './Summary';
