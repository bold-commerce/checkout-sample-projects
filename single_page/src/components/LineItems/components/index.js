export { default as LineItem } from './LineItem';
export { default as Product } from './Product';
