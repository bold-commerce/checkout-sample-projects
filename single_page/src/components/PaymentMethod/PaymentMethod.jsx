import { useBillingAddress, useBillingSameAsShipping, useCheckoutStore, usePaymentIframe, useShippingAddress, useShippingLines } from '@boldcommerce/checkout-react-components';
import React, { memo, useEffect, useState } from 'react';
import { CheckoutSection } from '../CheckoutSection';
import { EmptyState } from '../EmptyState';
import { LoadingState } from '../LoadingState';
import './PaymentMethod.css';
import { useTranslation } from 'react-i18next';

const PaymentMethod = ({ applicationLoading }) => {
  const { state } = useCheckoutStore();
  const { data: shippingAddress } = useShippingAddress();
  const { data: billingAddress } = useBillingAddress();
  const { data: billingSameAsShipping } = useBillingSameAsShipping();
  const { data: paymentIframe, loadingStatus, paymentIframeOnLoaded } = usePaymentIframe();
  const { data } = useShippingLines();
  const shippingLines = data.shippingLines;
  const orderStatus = state.orderInfo.orderStatus;
  const loading = (loadingStatus !== 'fulfilled' && orderStatus !== 'authorizing') || applicationLoading;

  return <MemoizedPaymentMethod
    billingAddress={billingAddress}
    shippingAddress={shippingAddress}
    shippingLines={shippingLines}
    billingSameAsShipping={billingSameAsShipping}
    paymentIframeUrl={paymentIframe.url}
    paymentIframeHeight={paymentIframe.height}
    onPaymentIframeLoaded={paymentIframeOnLoaded}
    loading={loading}
  />;
};

const MemoizedPaymentMethod = memo(({
  billingAddress,
  shippingAddress,
  shippingLines,
  billingSameAsShipping,
  paymentIframeUrl,
  paymentIframeHeight,
  onPaymentIframeLoaded,
  loading,
}) => {
  const [disabled, setDisabled] = useState();
  const { t } = useTranslation();

  useEffect(() => {
    if (billingSameAsShipping && Array.isArray(shippingAddress)) {
      setDisabled(true);
    } else if (!billingSameAsShipping && Array.isArray(billingAddress)) {
      setDisabled(true);
    } else if (shippingLines.length === 0) {
      setDisabled(true);
    } else {
      setDisabled(false);
    }
  }, [billingAddress, shippingAddress, billingSameAsShipping]);

  const style = {
    height: `${paymentIframeHeight}px`,
    display: (disabled || loading) ? 'none' : null,
  };

  let content = '';

  if (disabled) {
    content = <EmptyState title={t('payment.empty')} />;
  } else if (loading) {
    content = <LoadingState />;
  }

  return (
    <CheckoutSection
      className="FieldSet--PaymentMethod"
      title={t('payment.method')}
    >
      <iframe
        title="payments"
        data-bold-pigi-iframe
        className="PaymentMethod__Iframe"
        src={paymentIframeUrl}
        style={style}
        onLoad={onPaymentIframeLoaded}
      />
      { content }
    </CheckoutSection>
  );
});

export default PaymentMethod;
