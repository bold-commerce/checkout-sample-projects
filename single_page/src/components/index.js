export * from './Address';
export * from './BillingAddress';
export * from './CheckoutButton';
export * from './CheckoutSection';
export * from './Customer';
export * from './DiscountForm';
export * from './EmptyState';
export * from './Header';
export * from './LineItems';
export * from './LoadingState';
export * from './OrderErrors';
export * from './OrderSummary';
export * from './PaymentMethod';
export * from './ShippingAddress';
export * from './ShippingLines';
