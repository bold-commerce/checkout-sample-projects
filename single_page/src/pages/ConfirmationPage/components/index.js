export * from './ConfirmationList';
export * from './ConfirmationListItem';
