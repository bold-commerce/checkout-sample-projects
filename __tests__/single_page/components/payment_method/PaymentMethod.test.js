import React from 'react';
import { render } from "@testing-library/react";
import { PaymentMethod } from '../../../../single_page/src/components/PaymentMethod';
import {
  exampleUseBillingSameAsShipping as MOCKbillingSameAsShipping,
  exampleUseShippingAddress as MOCKshippingAddress,
  exampleUseBillingAddress as MOCKbillingAddress,
  exampleUseCheckoutStore as MOCKcheckoutStore,
  exampleUsePaymentIframe as MOCKpaymentIframe,
  exampleUseShippingLines as MOCKshippingLines
} from '../../../utils/hookHelpers';
import '../../../../single_page/src/i18n/config';

jest.mock('@boldcommerce/checkout-react-components', () => ({
  useBillingSameAsShipping: () => MOCKbillingSameAsShipping,
  useShippingAddress: () => MOCKshippingAddress,
  useBillingAddress: () => MOCKbillingAddress,
  usePaymentIframe : () => MOCKpaymentIframe,
  useShippingLines: () => MOCKshippingLines,
  useCheckoutStore: () => MOCKcheckoutStore
}));

describe('PaymentMethod', () => {
  test('renders PaymentMethod component', () => {
    const { asFragment } = render( <PaymentMethod /> );

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders PaymentMethod component while loading', () => {
    const { asFragment } = render( <PaymentMethod applicationLoading /> );

    expect(asFragment()).toMatchSnapshot();
  });
});
