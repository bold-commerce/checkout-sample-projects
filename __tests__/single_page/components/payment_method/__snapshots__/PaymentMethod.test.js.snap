// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethod renders PaymentMethod component 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--PaymentMethod"
  >
    <div
      class="FieldSet__Header"
    >
      <h3
        class="FieldSet__Heading"
      >
        Payment method
      </h3>
    </div>
    <div
      class="FieldSet__Content"
    >
      <iframe
        class="PaymentMethod__Iframe"
        data-bold-pigi-iframe="true"
        src="test.url"
        style="height: 21px;"
        title="payments"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`PaymentMethod renders PaymentMethod component while loading 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--PaymentMethod"
  >
    <div
      class="FieldSet__Header"
    >
      <h3
        class="FieldSet__Heading"
      >
        Payment method
      </h3>
    </div>
    <div
      class="FieldSet__Content"
    >
      <iframe
        class="PaymentMethod__Iframe"
        data-bold-pigi-iframe="true"
        src="test.url"
        style="height: 21px; display: none;"
        title="payments"
      />
      <div
        class="FieldSet--LoadingState"
      >
        <span
          class="stx-loading-spinner"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
